### PR TITLE
attr (new formula)

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/attr.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  homepage "https://savannah.nongnu.org/projects/attr"
+  url "https://mirror.csclub.uwaterloo.ca/nongnu/attr/attr-2.4.48.tar.gz"
+  sha256 "5ead72b358ec709ed00bbf7a9eaef1654baad937c001c044fe8b74c57f5324e7"
+  license "GPL-2.0-or-later"
+
+  depends_on "gettext" => :build
+  depends_on :linux
+
+  def install
+    # Fix No rule to make target `../libattr/libattr.la', needed by `attr'
```
Comment (MikeMcQuaid) [inline Formula/attr.rb:12]: Has this been submitted upstream? If not: can it be, please.

File: Formula/attr.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    system "./configure",
+           "--disable-debug",
+           "--disable-dependency-tracking",
+           "--disable-silent-rules",
+           "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system bin/"attr", "-l", "/bin/sh"
```
Comment (SMillerDev) [inline Formula/attr.rb:24]: Can this check output

**Review by SMillerDev — DISMISSED** [review]

File: Formula/attr.rb (hunk 1 above)
Comment (iMichka) [reply Formula/attr.rb:12]: I ran a test build on linux and it looks like it's not needed anymore:
https://github.com/Homebrew/linuxbrew-core/pull/21359

File: Formula/attr.rb (hunk 2 above)
Comment (iMichka) [reply Formula/attr.rb:24]: Yes, done

**Comment by iMichka** [comment]
The linux failure is expected as `ncurses` is not ready to be built for linux in this repo.

I ran a full linux build here, which was green: https://github.com/Homebrew/linuxbrew-core/pull/21359

**Comment by SMillerDev** [comment]
If it doesn't build successfully on Linux and is incompatible with macOS I think this one should wait for ncurses.

**Comment by iMichka** [comment]
It does build on Linux, see https://github.com/Homebrew/linuxbrew-core/pull/21359

To make it work in this repo:
- I need to get https://github.com/Homebrew/brew/pull/8907 done
- I need a new brew release
- I need to backport the ncurses changes to this repo

We can wait for this to happen, or just merge this as-is.

**Comment by SMillerDev** [comment]
I'd rather wait for that. There's no rush to get this merged and that way we have green builds here too. Otherwise it'd be very difficult to judge people making PRs for it.

**Comment by iMichka** [comment]
The `ncurses` changes have been merged.

It looks like `brew test-bot --only-formulae` does not install the needed dependencies though.

**Comment by iMichka** [comment]
@SeekingMeaning I was able to compile this by removing the `--with-included-gettext` flag from the gettext formula. This has been the case for the linux formula since years. I do not understand why this helps though.

**Comment by SeekingMeaning** [comment]
This might help to figure out the root cause: https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined

**Comment by iMichka** [comment]
Thanks @SeekingMeaning for finding this out. I am not sure what exactly is going on here, but thanks for the fix!

**Comment by SeekingMeaning** [comment]
Oops, this should have been squashed https://github.com/Homebrew/homebrew-core/commit/2a3d13b2cf89a95583687290e0e26d670ad0d002

**Pull request merged**